spark/v3.5/spark/src/test/java/org/apache/iceberg/spark/data/TestHelpers.java

```diff
@@ -804,11 +805,7 @@ public static List<DataFile> dataFiles(Table table) {
   }
 
   public static List<DataFile> dataFiles(Table table, String branch) {
-    TableScan scan = table.newScan();
-    if (branch != null) {
-      scan.useRef(branch);
-    }
-
+    TableScan scan = table.newScan().useRef(branch != null ? branch : SnapshotRef.MAIN_BRANCH);
```

Comment:
While I think it is better not to call `useRef` in the Spark code, it is not an option to always call `useRef` here.

It is critical that tests exercise paths that do not call that method.

Reply:
That's reasonable; I was mostly looking to just consolidate the code via useRef for both main and non-main cases, but it's important to make sure that we don't mask any issues in default table scan APIs. Removed the `useRef` calls for main.